pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -154,19 +157,27 @@ func (r *DNSPolicyReconciler) reconcileResources(ctx context.Context, dnsPolicy
 		return err
 	}
 
-	if err := r.reconcileDNSRecords(ctx, dnsPolicy, gatewayDiffObj); err != nil {
+	if err = r.reconcileDNSRecords(ctx, dnsPolicy, gatewayDiffObj); err != nil {
+		gatewayCondition = buildGatewayCondition(conditions.DNSPolicyReasonInvalid, err)
+		_ = r.updateGatewayCondition(ctx, gatewayCondition, gatewayDiffObj)
 		return err
 	}
 
-	if err := r.reconcileHealthChecks(ctx, dnsPolicy, gatewayDiffObj); err != nil {
+	if err = r.reconcileHealthChecks(ctx, dnsPolicy, gatewayDiffObj); err != nil {
+		gatewayCondition = buildGatewayCondition(conditions.DNSPolicyReasonInvalid, err)
+		_ = r.updateGatewayCondition(ctx, gatewayCondition, gatewayDiffObj)
```

Comment:
same here and below. Not much point in returning an error if all we do is ignore it